client/src/app/components/page/task-detail/task-detail.component.html

```diff
@@ -1,27 +1,35 @@
-<div class="card mt-4 container flex-row">
+<div class="mt-4 container border rounded-lg">
     <div class="row">
         <div class="colCard col-sm">
             <form [formGroup]="taskForm" (submit)="onSubmit()">
                 <h5 class="font-weight-light text-center text-lg-left mt-2 mb-2">Title</h5>
                 <input formControlName="title" class="form-control">
                 <h5 class="mt-2">Description</h5>
                 <input formControlName='description' class="form-control"><br>
-                <button class="btn btn-outline-success my-2 my-sm-0 w-50">
-                    {{buttonText}}</button>
             </form>
-            <button (click)="deleteTask()" *ngIf="updateMode" class="btn btn-danger mt-2 w-50">Delete</button>
+            <h5 class="mt-2">Choose your tags</h5>
+            <select2 [data]="allUniqueTags" (valueChanged)="onTagsChanged($event)" [width]="200" [value]="selectValue"
+                [options]="selectOptions"></select2>
         </div>
         <div class="colCard col-sm">
             <h5 class="mt-2">Deadline</h5>
             <app-date-picker *ngIf="fetchedData || !updateMode" [pickedDate]="taskForm.value.deadline"
```

Comment:
You don't need `fetchedData` flag here. You can just pass null value when data has no arrived from subscription.

You can also use handy contruction in angular templates:
taskForm && taskForm.value && taskForm.value.deadline -> taskForm?.value?.deadline

so when there is no object or there is an object but without properties you are looking for, `null` will be returned.

Reply:
I removed fetchedData but can't figure out how to do it without updateMode flag.

Reply:
Please remove conditional rendering, and use `taskForm?.value?.deadline` construct to pass initial value. Then inside component look for any changes with `ngOnChanges` lifecycle hook. Then inside `app-date-picker` component, when changing form value, use appropriate method of FormControl to update value.

Reply:
Removed conditional rendering from `<app-date-picker>` in task.detail file and removed conditional rendering from <ngb-timepicker> in datepicker file. Now date is updated in the ngOnChanges function.